src/db/sysdb_ops.c

```diff
@@ -2710,11 +2710,6 @@ int sysdb_store_user(struct sss_domain_info *domain,
         }
     }
 
-    if (pwd && (domain->legacy_passwords || !*pwd)) {
-        ret = sysdb_attrs_add_string(attrs, SYSDB_PWD, pwd);
-        if (ret) goto done;
-    }
-
```

Comment:
Please, do not remove this code entirely. Only the check for domain->legacy_passwords should be removed.
So, you'll end up with something like:
```
if (pwd && !*pwd) {
   ...
}
```